tests/unit/utils/Potrace-test.js

```diff
@@ -22,11 +22,12 @@ describe('Potrace.trace', () => {
     trace(testCanvas, 1)
       .then(resolve => {
         expect(resolve).to.equal(path)
+        done()
       })
-      .finally(done())
```

Comment:
@Bronson-Brown-deVost : FYI, here's the reason why the Potrace test issues are coming up just now. This should be
```javascript
.finally(done)
// or
finally(() => done())
```
The current form pretty much means that the test case is completed immediately (before even the promise runs). But, because the Potrace code IS executed, Istanbul picks that execution up. It just happens to execute _after_ the test case exits ... which makes Mocha/Karma think the test case is successful. Hence why we had passing tests + good code coverage but, unfortunately, broken tests. Sorry I didn't catch this one sooner!

Reply:
Thanks for catching that one.  Good to know about the promises, I guess it could become even worse if the test somehow passes and becomes "evergreen".

Also, there should be a few Istanbul "ignores" in the Potrace.js file for few sections of code that I could not figure out how to get the test to pass through.  Otherwise the file should be mostly testable.

Reply:
Ok, thanks, and I was reminded by looking at the Potrace code that you had some modifications to the original—so probably best to keep the tests. Here is the failure of the test I was seeing (I'm afraid it does seem they were "evergreen"):

```
1) should return an SVG path for a canvas
     Potrace.trace
     AssertionError: expected 'M1 15.5 L 1 30 1.518 30 L 2.036 30 11.848 25.076 L 21.659 20.153 23.242 22.826 L 24.826 25.5 27.328 27.844 L 29.83 30.189 31.92 29.525 L 34.011 28.862 37.005 26.342 L 40 23.822 40 21.396 L 40 18.971 36.75 16.989 L 33.5 15.007 31 15.039 L 28.5 15.071 26 16.5 L 23.5 17.929 22.25 17.965 L 21 18 21 13.455 L 21 8.909 23.25 6.705 L 25.5 4.5 28 2.802 L 30.5 1.105 15.75 1.052 L 1 1 1 15.5 ' to equal 'M1 15.5 L 1 30 1.518 30 L 2.036 30 11.848 25.076 L 21.659 20.153 23.242 22.826 L 24.826 25.5 27.328 27.844 L 29.83 30.189 31.92 29.525 L 34.011 28.862 37.005 26.342 L 40 23.822 40 21.396 L 40 18.971 36.75 16.989 L 33.5 15.007 31.2 15.004 L 28.901 15 25.921 16.541 L 22.941 18.082 21.971 17.482 L 21 16.882 21 12.896 L 21 8.909 23.256.705 L 25.5 4.5 28 2.802 L 30.5 1.105 15.75 1.052 L 1 1 1 15.5 '
```

Reply:
I'm happy if you want to check out this branch and push updates to get those truly green :)

Reply:
I just pushed in my latest version of Potrace.js, and I see it still fails in TravisCI, but only with Firefox.  I left the Potrace-test.js as is, I see you caught the problem with testing the Error object as well.  I ran the debugger and see that the values returned by Potrace.js is exactly what we expect (hence Chrome testing positive).  So the problem seems to lie with Firefox and the unit test itself.  Any ideas?